File: dotnet/DesignPatterns/BehavioralDesignPatterns/MediatorDesignPattern/MediatorDesignPattern.Example3/Bidder.cs
namespace MediatorDesignPattern.Example3;

/// <summary>
/// Colleague abstract class - Base class for all bidders
/// </summary>
public abstract class Bidder
{
    protected IAuctionMediator Mediator;
    public string Name { get; }

    protected Bidder(IAuctionMediator mediator, string name)
    {
        Mediator = mediator;
        Name = name;
    }

    public abstract void PlaceBid(decimal amount);
    public abstract void ReceiveNotification(string message);
}

File: nodejs/DesignPatterns/BehavioralDesignPatterns/MediatorDesignPattern/MediatorDesignPattern.Example3/src/Bidder.ts
import { IAuctionMediator } from './IAuctionMediator';

export abstract class Bidder {
  protected mediator: IAuctionMediator;
  readonly name: string;

  constructor(mediator: IAuctionMediator, name: string) {
    this.mediator = mediator;
    this.name = name;
  }

  abstract placeBid(amount: number): void;
  abstract receiveNotification(message: string): void;
}
